test/integration/apiserver/field_validation_test.go

```diff
@@ -0,0 +1,681 @@
+		bodyFile string
+		resource string
+		absPath  string
+	}{
+		{
+			name:     "ignore-validation-deployment",
+			params:   map[string]string{"fieldValidation": "Ignore"},
+			bodyFile: "./testdata/deploy-small.json",
+			resource: "deployments",
+			absPath:  appsPath,
+		},
+		{
+			name:     "strict-validation-deployment",
+			params:   map[string]string{"fieldValidation": "Strict"},
+			bodyFile: "./testdata/deploy-small.json",
+			resource: "deployments",
+			absPath:  appsPath,
+		},
+		{
+			name:     "ignore-validation-pod",
+			params:   map[string]string{"fieldValidation": "Ignore"},
+			bodyFile: "./testdata/pod-medium.json",
+			resource: "pods",
+			absPath:  corePath,
+		},
+		{
+			name:     "strict-validation-pod",
+			params:   map[string]string{"fieldValidation": "Strict"},
+			bodyFile: "./testdata/pod-medium.json",
+			resource: "pods",
+			absPath:  corePath,
+		},
+		{
+			name:     "ignore-validation-big-pod",
+			params:   map[string]string{"fieldValidation": "Ignore"},
+			bodyFile: "./testdata/pod-large.json",
+			resource: "pods",
+			absPath:  corePath,
+		},
+		{
+			name:     "strict-validation-big-pod",
+			params:   map[string]string{"fieldValidation": "Strict"},
+			bodyFile: "./testdata/pod-large.json",
+			resource: "pods",
+			absPath:  corePath,
+		},
+	}
+
+	for _, bm := range benchmarks {
+		b.Run(bm.name, func(b *testing.B) {
+			b.ResetTimer()
+			b.ReportAllocs()
+			for n := 0; n < b.N; n++ {
+				// append the timestamp to the name so that we don't hit conflicts when running the test multiple times
+				// (i.e. without it -count=n for n>1 will fail, this might be from not tearing stuff down properly).
+				bodyBase, err := os.ReadFile(bm.bodyFile)
+				if err != nil {
+					b.Fatal(err)
+				}
+
+				objName := fmt.Sprintf("obj-%s-%d-%d-%d", bm.name, n, b.N, time.Now().UnixNano())
+				objString := fmt.Sprintf(string(bodyBase), fmt.Sprintf(`"%s"`, objName))
+				body := []byte(objString)
+
+				postReq := client.CoreV1().RESTClient().Post().
+					AbsPath(bm.absPath).
+					Namespace("default").
+					Resource(bm.resource)
+				for k, v := range bm.params {
+					postReq = postReq.Param(k, v)
+				}
+
+				_, err = postReq.Body(body).
+					DoRaw(context.TODO())
+				if err != nil {
+					b.Fatal(err)
+				}
+
+				// TODO: put PUT in a different bench case than POST (ie. have a baseReq) be a part of the test case.
+				putReq := client.CoreV1().RESTClient().Put().
+					AbsPath(bm.absPath).
+					Namespace("default").
+					Resource(bm.resource).
+					Name(objName)
+				for k, v := range bm.params {
+					putReq = putReq.Param(k, v)
+				}
+
+				_, err = putReq.Body(body).
+					DoRaw(context.TODO())
+				if err != nil {
+					b.Fatal(err)
+				}
+			}
+		})
+
+	}
+}
+
+// smpTestSetup applies an object that will later be patched
+// in the actual test/benchmark.
+func smpTestSetup(t testing.TB, client clientset.Interface) {
+	bodyBase, err := os.ReadFile("./testdata/deploy-small.json")
```

Comment:
I'd also ideally avoid having files read from disk for tests, since it makes it harder to understand what the test does.